test/data/css/tests.js

```diff
@@ -220,13 +220,20 @@ exports.test_data = {
             separator_output: '\\n\\n',
         }, {
             options: [
+                { name: "preserve_newlines", value: "true" }
+            ],
+            separator_input: '\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n',
+            separator_output: '\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n',            
+        }, {
+            options: [
                 { name: "preserve_newlines", value: "false" }
             ],
             separator_input: '\\n\\n',
             separator_output: '\\n',
         }],
         tests: [
             { input: '.div {}{{separator_input}}.span {}', output: '.div {}{{separator_output}}.span {}' },
+            { input: '.tool-tip {\n\tposition: relative;{{separator_input}}\t.tool-tip-content {\n\t\t&>* {\n\t\t\tmargin-top: 0;\n\t\t}{{separator_input}}\t\t.mixin-box-shadow(.2rem .2rem .5rem rgba(0, 0, 0, .15));\n\t\tpadding: 1rem;\n\t\tposition: absolute;\n\t\tz-index: 10;\n\t}\n}', output: '.tool-tip {\n\tposition: relative;{{separator_output}}\t.tool-tip-content {\n\t\t&>* {\n\t\t\tmargin-top: 0;\n\t\t}{{separator_output}}\t\t.mixin-box-shadow(.2rem .2rem .5rem rgba(0, 0, 0, .15));\n\t\tpadding: 1rem;\n\t\tposition: absolute;\n\t\tz-index: 10;\n\t}\n}' },
```

Comment:
@TimvdEijnden 
Could you put these long 'input' and `output` members on separate lines for readability?  

Still need more tests:
What about the `\n` at `{\n\tposition:`?  If we put multiple newlines there will they be preserved?

What about test inputs that have no new lines and make sure the appropriate newlines are still added when `preserve_newlines` is `true` or 'false`.  You'll need to split `separator_output` into to two settings one where a newline should be added and one where it shouldn't.

Also, how does this interact with `selector_separator_newline`?  

(I know your code won't have this problem, but part of the point of the tests is to guard future changes.)